src/c/dune

```diff
@@ -26,31 +31,78 @@
 (data_only_dirs vendor)
 
 (rule
+ (targets libraylib.a dllraylib.so)
+ (enabled_if
+  (or
+   (= %{system} linux)
+   (= %{system} linux_elf)))
  (deps
-  (source_tree vendor))
+  (source_tree vendor)
+  linux.patch)
+ (action
+  (no-infer
+   (progn
+    (run chmod +w vendor/src/config.h)
+    (with-stdin-from
+     linux.patch
+     (run patch --binary -p1))
+    (run make -C vendor/src RAYLIB_LIBTYPE=STATIC -j 8)
+    (copy vendor/src/libraylib.a libraylib.a)
+    (run make -C vendor/src clean)
+    (run make -C vendor/src RAYLIB_LIBTYPE=SHARED -j 8)
+    (copy vendor/src/libraylib.so.3.0.0 dllraylib.so)))))
+
+(rule
  (targets libraylib.a dllraylib.so)
+ (enabled_if
+  (= %{system} macosx))
+ (deps
+  (source_tree vendor))
+ (action
+  (no-infer
+   (progn
+    (run make -C vendor/src RAYLIB_LIBTYPE=STATIC -j 8)
+    (copy vendor/src/libraylib.a libraylib.a)
+    (run make -C vendor/src clean)
+    (run make -C vendor/src RAYLIB_LIBTYPE=SHARED -j 8)
+    (copy vendor/src/libraylib.3.0.0.dylib dllraylib.so)))))
+
+(rule
+ (targets libraylib.a dllraylib.dll)
+ (enabled_if
+  (= %{system} mingw64))
+ (deps
+  (source_tree vendor)
+  mingw64.patch)
+ (action
+  (no-infer
+   (progn
+    (run chmod +w vendor/src/Makefile)
+    (with-stdin-from
+     mingw64.patch
+     (run patch --binary -p1))
+    (run make -C vendor/src RAYLIB_LIBTYPE=STATIC -j 8)
+    (copy vendor/src/libraylib.a libraylib.a)
+    (run make -C vendor/src clean)
+    (run make -C vendor/src RAYLIB_LIBTYPE=SHARED -j 8)
+    (copy vendor/src/raylib.dll dllraylib.dll)))))
+
+(rule
+ (targets libraylib.a dllraylib.dll)
```

Comment:
I've tried to build it on FreeBSD in a VM. The shared library here needs to be dllraylib.so instead of dllraylib.dll

Apart from that, I could compile it. (I couldn't like against GL, but I think that's due to the VM)